main.go

```diff
@@ -27,6 +27,8 @@ import (
 	"k8s.io/klog/klogr"
 	clusterv1alpha2 "sigs.k8s.io/cluster-api/api/v1alpha2"
 	clusterv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	kubeadmv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha2"
```

Comment:
This rename makes me think this is related to the kubeadm types and not the bootstrap provider types.

Reply:
Fixed, thanks!